pkg/investigations/insightsoperatordown/insightsoperatordown.go

```diff
@@ -0,0 +1,140 @@
+// SOP https://github.com/openshift/ops-sop/blob/master/v4/troubleshoot/clusteroperators/insights.md
+// Step: Check for banned user
+// Step: Check for https://issues.redhat.com/browse/OCPBUGS-22226
+//   - Reproducer: block console.redhat.com via rule group in aws account
+package insightsoperatordown
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	configv1 "github.com/openshift/api/config/v1"
+	investigation "github.com/openshift/configuration-anomaly-detection/pkg/investigations/investigation"
+	k8sclient "github.com/openshift/configuration-anomaly-detection/pkg/k8s"
+	"github.com/openshift/configuration-anomaly-detection/pkg/logging"
+	"github.com/openshift/configuration-anomaly-detection/pkg/networkverifier"
+	"github.com/openshift/configuration-anomaly-detection/pkg/notewriter"
+	"github.com/openshift/configuration-anomaly-detection/pkg/ocm"
+	"k8s.io/apimachinery/pkg/fields"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Investigation struct{}
+
+func (c *Investigation) Run(r *investigation.Resources) (investigation.InvestigationResult, error) {
+	result := investigation.InvestigationResult{}
+	notes := notewriter.New(r.Name, logging.RawLogger)
+
+	user, err := ocm.GetCreatorFromCluster(r.OcmClient.GetConnection(), r.Cluster)
+	if err != nil {
+		notes.AppendWarning("encountered an issue when checking if the cluster owner is banned: %s\nPlease investigate.", err)
+		return result, r.PdClient.EscalateIncidentWithNote(notes.String())
+	}
+
+	if user.Banned() {
+		notes.AppendWarning("User is banned: %s", user.BanCode())
+		notes.AppendWarning("Ban description: %s", user.BanDescription())
+		notes.AppendWarning("Please open a proactive case, so that MCS can resolve the ban or organize a ownership transfer.")
+	} else {
+		notes.AppendSuccess("User is not banned.")
+	}
+
+	// We continue with the next step OCPBUG22226 even if the user is banned.
+	k8scli, err := k8sclient.New(r.Cluster.ID(), r.OcmClient, r.Name)
+	if err != nil {
+		return result, fmt.Errorf("unable to initialize k8s cli: %w", err)
+	}
+	defer func() {
+		deferErr := k8sclient.Cleanup(r.Cluster.ID(), r.OcmClient, r.Name)
+		if deferErr != nil {
+			logging.Error(deferErr)
+			err = errors.Join(err, deferErr)
+		}
+	}()
+
+	coList := &configv1.ClusterOperatorList{}
+	listOptions := &client.ListOptions{FieldSelector: fields.SelectorFromSet(fields.Set{"metadata.name": "insights"})}
+	err = k8scli.List(context.TODO(), coList, listOptions)
+	if err != nil {
+		return result, fmt.Errorf("unable to list insights clusteroperator: %w", err)
+	}
+
+	if len(coList.Items) != 1 {
+		return result, fmt.Errorf("found %d clusteroperators, expected 1", len(coList.Items))
+	}
+	co := coList.Items[0]
+
+	if isOCPBUG22226(&co) {
+		notes.AppendWarning("Found symptom of OCPBUG22226. Try deleting the insights operator pod to remediate.")
+		notes.AppendWarning("$ oc -n openshift-insights delete pods -l app=insights-operator --wait=false")
```

Comment:
Would merge the warning here into one with a newline. Every warning should be a single issue, once we stack issues this might cause confusion.